src/Tools/BuildValidator/BuildConstructor.cs

```diff
@@ -302,6 +307,14 @@ private static VisualBasicCompilationOptions CreateVisualBasicCompilationOptions
                 publicSign: false,
                 reportSuppressedDiagnostics: false,
                 metadataImportOptions: MetadataImportOptions.Public);
+            compilationOptions.DebugPlusMode = plus;
+
+            return compilationOptions;
+
+            string? OptionToString(string option) => pdbCompilationOptions.TryGetUniqueOption(option, out var value) ? value : null;
+            bool? OptionToBool(string option) => pdbCompilationOptions.TryGetUniqueOption(option, out var value) ? ToBool(value) : null;
+            T? OptionToEnum<T>(string option) where T : struct => pdbCompilationOptions.TryGetUniqueOption(option, out var value) && Enum.TryParse<T>(value, out var enumValue) ? enumValue : null;
```

Comment:
OptionToEnum uses `&&` with two Try* methods in sequence, but OptionToBool uses one Try* method then calls into ToBool. Would be good to make them consistent.